### PR TITLE
Make TrussServer single-worker, remove faulty middleware.

### PR DESCRIPTION
**Pull request by marius-baseten**
## :rocket: What
* Remove support for multiple server processes. Potential performance bottlenecks should be addressed differently. Some truss examples will need to be updated (mistral &co).
* Remove faulty termination middleware that broke request disconnects to propagate.
* Assert safe shutdown behavior in prediction integration test (grace period 120).

This subsumes closed #1166

### CODE REVIEW

File: truss/tests/test_model_inference.py [hunk 1]
```diff
@@ -760,6 +768,85 @@ def _test_invocations(expected_code):
...
 
+def _patch_termination_timeout(container: Container, truss_container_fs):
+    app_path = truss_container_fs / "app"
+    sys.path.append(str(app_path))
+    import truss_server
+
+    local_server_source = pathlib.Path(truss_server.__file__)
+    container_server_source = "/app/truss_server.py"
+    modified_content = local_server_source.read_text().replace(
+        "TIMEOUT_GRACEFUL_SHUTDOWN = 120", "TIMEOUT_GRACEFUL_SHUTDOWN = 3"
```
[the remark this answers is not in the document]
Comment (marius-baseten) [reply truss/tests/test_model_inference.py:779]: That would be an alternative, but then I'd need to plumb setting evn vars through `tr.docker_run` or commit a new image...

This here is not pretty, but it works now and anything else will be extra work. If you want me to change it, let me know what exactly concerns you about it.

File: truss-chains/tests/itest_chain/itest_chain.py [hunk 2]
```diff
@@ -125,7 +125,7 @@ async def run_remote(
         for part in text_parts.parts:
             value += self._text_to_num.run_remote(part)
         return (
-            value,
+            value + 100000,
```
[the remark this answers is not in the document]
Comment (marius-baseten) [reply truss-chains/tests/itest_chain/itest_chain.py]: Not changed anymore, this was unrelated...

File: truss/tests/test_model_inference.py [hunk 3]
```diff
@@ -760,6 +768,85 @@ def _test_invocations(expected_code):
...
+
+        # Test starting two requests, each taking 2 seconds, then terminating server.
+        # They should both finish successfully since the server grace period is 120 s.
+        task_0 = asyncio.create_task(predict_request({"seconds": 2, "task": 0}))
+        await asyncio.sleep(0.1)  # Yield to event loop to make above task run.
+        task_1 = asyncio.create_task(predict_request({"seconds": 2, "task": 1}))
+        await asyncio.sleep(0.1)  # Yield to event loop to make above task run.
+
+        t0 = time.perf_counter()
+        container.stop(30)
```
[the remark this answers is not in the document]
Comment (marius-baseten) [reply truss/tests/test_model_inference.py:820]: Done.

File: truss/templates/server/truss_server.py [hunk 4]
```diff
@@ -331,29 +322,20 @@ def create_application(self):
...
-
-        termination_handler_middleware = TerminationHandlerMiddleware(
-            on_stop=lambda: None,
-            on_term=exit_self,
-        )
-        app.add_middleware(BaseHTTPMiddleware, dispatch=termination_handler_middleware)
         return app
 
     def start(self):
+        log_level = "DEBUG" if self._config["runtime"]["enable_debug_logs"] else "INFO"
```
Comment (squidarth) [inline truss/templates/server/truss_server.py:328]: awesome, thanks for adding this! @bdubayah will be very happy

**Review by squidarth — APPROVED** [review]

**Pull request merged**